p2p/dhtdiscovery/discovery.go

```diff
@@ -124,12 +122,30 @@ func AdvertiseForPeerDiscovery() Opt {
 	}
 }
 
+func WithAdvertiseRetryDelay(aint time.Duration) Opt {
+	return func(d *Discovery) {
+		d.advertiseRetryDelay = aint
+	}
+}
```

Comment:
How about using simply `value` for the argument name? The name of the function specifies precisely what it is (an advertise retry delay). You would avoid copy-paste mistakes as a benefit (`aint` makes no sense here) :wink:. 
```suggestion
func WithAdvertiseRetryDelay(value time.Duration) Opt {
	return func(d *Discovery) {
		d.advertiseRetryDelay = value
	}
}
```